test/IntegrationTests/TicketService/test_create_custom_fields.py

```diff
@@ -0,0 +1,287 @@
+        invalid_field_type = "INVALID_TYPE"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "invalid_field_type_test",
+            "field_type": invalid_field_type,
+            "required": True
+        }
+
+        # Attempt to add a custom field with an invalid field type
+        logger.info(f"Testing invalid field type with data: {data}")
+        response = requests.post(url, json=data)
+
+        # Assert that the API returns a 400 status code
+        self.assertEqual(response.status_code, HTTPStatus.BAD_REQUEST, "Invalid field type should result in 400")
+
+        # Expected error response
+        expected_message = {
+            "detail": f"Unsupported field type: {invalid_field_type}. Allowed types are: VARCHAR(255), INT, BOOLEAN, DATE, MEDIUMTEXT, FLOAT, TEXT"
+        }
+
+        # Convert the response to JSON for comparison
+        actual_response = response.json()
+
+        # Assert the entire response matches the expected structure
+        self.assertEqual(actual_response, expected_message, "Response does not match the expected structure")
+
+        logger.info("Test case for invalid field type passed.")
+
+    def test_add_custom_field_empty_field_name(self):
+        """Test adding a custom field with an empty field name."""
+        url = f"{self.BASE_URL}/custom_fields"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "",  # Empty field name
+            "field_type": "VARCHAR(255)",  # Valid field type
+            "required": True
+        }
+
+        # Attempt to add a custom field with an empty field name
+        logger.info(f"Testing empty field name with data: {data}")
+        response = requests.post(url, json=data)
+
+        # Assert that the API returns a 400 status code
+        self.assertEqual(response.status_code, HTTPStatus.BAD_REQUEST, "Empty field name should result in 400")
+
+        # Expected error response
+        expected_message = {
+            "detail": "Invalid field name ''. Field names can only contain letters, numbers, and underscores."
+        }
+
+        # Convert the response to JSON for comparison
+        actual_response = response.json()
+
+        # Assert the entire response matches the expected structure
+        self.assertEqual(actual_response, expected_message, "Response does not match the expected structure")
+
+        logger.info("Test case for empty field name passed.")
+
+    def test_add_custom_field_varchar(self):
+        self._test_add_custom_field("VARCHAR(255)")
+
+    def test_add_custom_field_int(self):
+        self._test_add_custom_field("INT")
+
+    def test_add_custom_field_boolean(self):
+        self._test_add_custom_field("BOOLEAN")
+
+    def test_add_custom_field_date(self):
+        self._test_add_custom_field("DATE")
+
+    def test_add_custom_field_mediumtext(self):
+        self._test_add_custom_field("MEDIUMTEXT")
+
```

Comment:
Can we add for all these fields?

["VARCHAR(255)", "INT", "BOOLEAN", "DATE", "MEDIUMTEXT", "FLOAT", "TEXT"]

Dont resolve this comment. Just reply to this comment once you fix it

Reply:
Yes sir we can add all those fields, i included test cases for remaining data types(float, text).